### PR TITLE
Allow windows to be invisible on opening.

### LINKED ISSUE #66: IsVisible value in WindowOptions isn't preserved when creating the window
# Summary
Some options, such as IsVisible, aren't applied when creating the window.

# Steps to reproduce
```cs
var opts = WindowOptions.Default;
opts.IsVisible = false;
var window = Window.Create(opts);
window.Run(); // window is visible
```

# Comments
https://discordapp.com/channels/521092042781229087/607634593201520651/646429503601639454

### PR DESCRIPTION
**Pull request by devvoid**
# Summary of the PR
Windows can now be opened invisible via setting IsVisible on WindowOptions.

# Related issues, Discord discussions, or proposals
Fixes #66 

# Further Comments

### CODE REVIEW

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 1]
```diff
@@ -380,6 +395,12 @@ public void Close()
             }
         }
 
+        /// <summary>
+        /// Open the window.
+        /// </summary>
+        /// <remarks>
+        /// This should be used if you want to setup your own loop; otherwise, you should use <see cref="WindowExtensions.Run"/> instead
+        /// </remarks>
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Was there not any documentation to inherit for this method? If not, these doc comments should be moved to the relevant interface in Windowing.Common instead

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 2]
```diff
@@ -420,6 +441,8 @@ public unsafe void Open()
                     _glfw.WindowHint(WindowHintClientApi.ClientApi, ClientApi.OpenGLES);
                     break;
             }
+            
+            _glfw.WindowHint(WindowHintBool.Visible, _initialOptions.IsVisible);
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Can you double check that all proprerties are used within this method in some way? Visibility might not be the only property we didn't send to GLFW...

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 3]
```diff
@@ -746,13 +772,28 @@ private unsafe void InitializeCallbacks()
...
         }
 
+        /// <summary>
+        /// Checks whether or not Vulkan is supported for this window.
+        /// </summary>
+        /// <remarks>
+        /// This doesn't guarantee that context or surface creation will succeed; consult the GLFW documentation for
+        /// more details.
+        /// </remarks>
         public bool IsVulkanSupported => _glfw.VulkanSupported();
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Was there not any documentation to inherit for this method? If not, these doc comments should be moved to the relevant interface in Windowing.Common instead

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 4]
```diff
@@ -746,13 +772,28 @@ private unsafe void InitializeCallbacks()
...
 
+        /// <summary>
+        /// Create a Vulkan surface.
+        /// </summary>
+        /// <param name="instance">The Vulkan instance to create a surface for.</param>
+        /// <param name="allocator">A custom Vulkan allocator. Can be omitted by passing null.</param>
+        /// <typeparam name="T">Allocator type</typeparam>
+        /// <returns>A handle to the Vulkan surface created</returns>
+        /// <exception cref="GlfwException">Thrown if GLFW fails to create a Vulkan surface</exception>
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Was there not any documentation to inherit for this method? If not, these doc comments should be moved to the relevant interface in Windowing.Common instead

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 5]
```diff
@@ -761,6 +802,11 @@ public unsafe VkHandle CreateSurface<T>(VkHandle instance, T* allocator)
             return surface[0];
         }
 
+        /// <summary>
+        /// Get the extensions required for Vulkan to work on this platform.
+        /// </summary>
+        /// <param name="count">The number of extensions in the returned array</param>
+        /// <returns>An array of strings, containing names for all required extensions</returns>
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Was there not any documentation to inherit for this method? If not, these doc comments should be moved to the relevant interface in Windowing.Common instead

File: src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs [hunk 6]
```diff
@@ -7,18 +7,37 @@
 
 namespace Silk.NET.Windowing.Common
 {
+    /// <summary>
+    /// Extensions for IWindow
+    /// </summary>
     public static class WindowExtensions
     {
+        /// <summary>
+        /// Invoke this Func on the window.
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs]:
```suggestion
        /// Executes a Func&lt;&gt; delegate on the given window.
```

File: src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs [hunk 7]
```diff
@@ -7,18 +7,37 @@
...
+    /// </summary>
     public static class WindowExtensions
     {
+        /// <summary>
+        /// Invoke this Func on the window.
+        /// </summary>
+        /// <param name="window">The window to invoke on.</param>
+        /// <param name="t">The function to invoke.</param>
+        /// <typeparam name="T">Type of the function.</typeparam>
+        /// <returns></returns>
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs]:
```suggestion
        /// <remarks>This function is not applicable on single-threaded windows</remarks>
        /// <returns>The return value of the given delegate.</returns>
```

File: src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs [hunk 8]
```diff
@@ -7,18 +7,37 @@
...
         public static T Invoke<T>(this IWindow window, Func<T> t)
         {
             return (T) window.Invoke(t);
         }
 
+        /// <summary>
+        /// Invoke this <see cref="Action"/> on the window.
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs]:
```suggestion
        /// Executes a <see cref="Action"/> on the given window.
```

File: src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs [hunk 9]
```diff
@@ -7,18 +7,37 @@
...
+        /// </summary>
+        /// <param name="window">The window to invoke on.</param>
+        /// <param name="t">The action to invoke.</param>
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs]:
```suggestion
        /// <param name="t">The action to invoke.</param>
        /// <remarks>This function is not applicable on single-threaded windows</remarks>
```

File: src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs [hunk 10]
```diff
@@ -7,18 +7,37 @@
...
         public static void Invoke(this IWindow window, Action t)
         {
             window.Invoke(t);
         }
 
+        /// <summary>
+        /// Start an event loop on this window.
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs]:
```suggestion
        /// Start the default event loop on this window.
```

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 11]
```diff
@@ -87,6 +86,9 @@ public GlfwWindow(WindowOptions options)
         /// <inheritdoc />
         public int RunningSlowTolerance { get; set; }
 
+        /// <summary>
+        /// Whether or not this window is queued to close.
+        /// </summary>
         public unsafe bool IsClosing => _glfw.WindowShouldClose(_windowPtr);
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Was there not any documentation to inherit for this method? If not, these doc comments should be moved to the relevant interface in Windowing.Common instead

**Comment by devvoid** [comment]
It's great to be back! I think that's everything, ready for re-review

**Comment by Perksey** [comment]
Waiting for CI to report green but this looks good to me.

Thanks again and welcome back :D

**Pull request merged**